assets/hip-1056/protobuf/services/contract_types.proto

```diff
@@ -0,0 +1,80 @@
+/**
+ * # Contract Message Types
+ * Message types used in contract transactions.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in
+ * [RFC2119](https://www.ietf.org/rfc/rfc2119) and clarified in
+ * [RFC8174](https://www.ietf.org/rfc/rfc8174).
+ */
+syntax = "proto3";
+
+package proto;
+
+// SPDX-License-Identifier: Apache-2.0
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.contract">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "services/basic_types.proto";
+import "google/protobuf/wrappers.proto";
+
+/**
+ * Context of an internal call in an EVM transaction that is not otherwise externalized.<br/>
+ */
+message InternalCallContext {
+    /**
+     * The gas available for the call.<br/>
+     */
+    uint64 gas = 1;
+
+    /**
+     * The value sent with the call.<br/>
+     */
+    uint64 value = 2;
+
+    /**
+     * The call data for the call.<br/>
+     */
+    bytes call_data = 3;
+}
+
+/**
+ * Results of executing a EVM transaction.<br/>
+ */
+message EvmTransactionResult {
+    /**
+     * The Hedera id of the caller.<br/>
+     */
+    AccountID sender_id = 1;
```

Comment:
Q: Do `sender_id` and `contract_id` need to be here, don't we already have them on input?
Could they be in `InternalCallContext` assuming they are only necessary for an internal call.
In all regular EVM calls both here items are in the transaction body explicitly or RLP encoded

Reply:
We _could_ shift the responsibility to stream consumer to resolving aliases to `0.0.X` ids, but this is just a vanishing detail in the big picture, and not worth it imo.